### PR TITLE
Click data menu items to set as reference data

### PR DESCRIPTION
**Pull request by bmorris3**
For https://github.com/spacetelescope/jdaviz/pull/2179.

For now, I'm exposing which item is selected as reference data by adding an asterisk to the item label. This is not a permanent solution, and here as a placeholder until we discuss with a long-term solution with @Jenneh.

https://github.com/bmorris3/jdaviz/assets/3497584/7b8b531c-a225-4064-ae3f-5c72ba361126

### CODE REVIEW

**Comment by bmorris3** [comment]
Latest update: we now have "notification badges" (small red-orange dots in the upper right) for data menu items that are selected as reference data. This preserves horizontal space.

https://github.com/bmorris3/jdaviz/assets/3497584/7bc1897f-8560-4c41-bdbc-a50e43be6651

File: jdaviz/components/viewer_data_select_item.vue [hunk 1]
```diff
@@ -11,29 +11,28 @@
...
-    <j-tooltip :tooltipcontent="'data label: '+item.name" span_style="font-size: 12pt; padding-top: 6px; padding-left: 6px; width: calc(100% - 80px); white-space: nowrap; cursor: default;">
-      <j-layer-viewer-icon span_style="margin-left: 4px; margin-right: 2px" :icon="icon" color="#000000DE"></j-layer-viewer-icon>      
-      <div class="text-ellipsis-middle" style="font-weight: 500">
+    <j-tooltip
+      :tooltipcontent="isRefData() ? 'Reference data' : 'Click to set as reference data'"
+      span_style="font-size: 12pt; padding-top: 6px; width: calc(100% - 80px); white-space: nowrap; cursor: default;">
+    <span
+      style="cursor: pointer;"
+      @click="selectRefData"
+    >
```
Comment (kecnry) [inline jdaviz/components/viewer_data_select_item.vue]: can we easily skip the tooltip and click entirely for viewers that don't support reference data (scatter viewers in lcviz, for example 😬 )?  Right now the tooltip and pointer cursor appear - which isn't ideal - but at least clicking is gracefully ignored without any tracebacks.

**Comment by pllim** [comment]
I dunno if this has been discussed or not while I was out, but when I change the reference data, the zoom resets. Is this something we just tell the user to live with it, or is it too annoying?

**Comment by pllim** [comment]
FWIW, the click to set reference worked as expected for me except for zoom reset. 👏

**Comment by rosteen** [comment]
I think I don't have a clear idea anymore of the motivation here (I haven't been tuned in to all the efforts going on that needed this), but my main concern is that "reference data" isn't really meaningful to users, and thus exposing something that says "click here to set reference data" may not be useful. I'm also not a big fan of the orange dot, but it works as a placeholder and this isn't getting merged into main so my comments don't need to be blockers.

**Review by pllim — APPROVED** [review]
Works for me but I agree with others that the wordings can be more user-friendly. Thanks!

File: jdaviz/components/viewer_data_select_item.vue [hunk 2]
```diff
@@ -13,7 +13,7 @@
     </div>
 
     <j-tooltip
-      :tooltipcontent="isRefData() ? 'Reference data' : 'Click to set as reference data'"
+      :tooltipcontent="isRefData() && viewer.config === 'imviz' ? 'Current viewer orientation' : 'Set viewer orientation'"
```
Comment (bmorris3) [inline jdaviz/components/viewer_data_select_item.vue]: @rosteen @pllim I've revised the tooltips with this language.

~@kecnry This tooltip won't be applied outside of Imviz.~ update: improvements in [b7a8fa2](https://github.com/bmorris3/jdaviz/pull/5/commits/b7a8fa2a4f54293fd31eaae69a27fb6b5692f39b)

File: jdaviz/components/viewer_data_select_item.vue (hunk 1 above)
Comment (bmorris3) [reply jdaviz/components/viewer_data_select_item.vue]: Done: https://github.com/bmorris3/jdaviz/pull/5#discussion_r1228452362

**Review by kecnry — APPROVED** [review]
Looks good now!

**Pull request merged**